test/features/build_server/compiler_interface_test.dart

```
@@ -0,0 +1,319 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:flyde/core/async/connect.dart';
+import 'package:flyde/core/async/interface.dart';
+import 'package:flyde/core/fs/compiler/installed_compiler.dart';
+import 'package:flyde/core/fs/configs/compiler_config.dart';
+import 'package:flyde/core/fs/file_extension.dart';
+import 'package:flyde/core/fs/search_directory.dart';
+import 'package:flyde/core/fs/wrapper/source_file.dart';
+import 'package:flyde/core/networking/protocol/compile_status.dart';
+import 'package:flyde/features/build_server/cache/project_cache.dart';
+import 'package:flyde/features/build_server/compiler_interface.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+import '../../helpers/clear_test_cache_directory.dart';
+import '../../helpers/create_dummy_project_cache.dart';
+import '../../helpers/value_hook.dart';
+
+Future<List<CompileStatusMessage>> _buildProject(
+  MainInterface interface,
+  List<SourceFile> files,
+  Map<String, String> fileMap,
+  CompilerConfig config,
+  ProjectCache cache, {
+  bool skipUpdateFiles = false,
+}) async {
+  final Completer<List<CompileStatusMessage>> completer = Completer();
```

Why not `VHook`? This has the posibility to block forever (or at least 30sec in test cases)

---

In this case `VHook` has no complete error method or something like that. Just ignoring the error and letting the hook time out would drop information. Printing the error message doesn't feel right either

---

Should we add this to `VHook`? That as soon as you await a value the error is thrown?

---

I think we should resolve this and open a PR for VHook. [https://github.com/Coronon/flyde/pull/9#issuecomment-914277597](https://github.com/Coronon/flyde/pull/9#issuecomment-914277597)